File: src/test/java/com/example/probation/repository/RoleRepositoryTest.java
package com.example.probation.repository;

import com.example.probation.core.entity.Role;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.jdbc.Sql;

import static org.junit.jupiter.api.Assertions.*;

@Sql({"/init.sql"})
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class RoleRepositoryTest {
    @Autowired
    private RoleRepository repository;

    @Test
    public void getRoleByRoleName() {
        final String role = "USER";
        final Role userRole = repository.getRoleByRoleName(role);
        assertNotNull(userRole);
        assertEquals(userRole.getRoleName(), role);
    }
}

File: src/test/kotlin/com/example/probation/repository/RoleRepositoryTest.kt
package com.example.probation.repository

import com.example.probation.core.enums.Roles
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired


class RoleRepositoryTest(
    @Autowired
    private val repository: RoleRepository
) : RepositoryTest() {

    @Test
    fun getRoleByRoleName() {
        val role = Roles.USER.role
        repository.getRoleByRoleName(role)?.apply {
            Assertions.assertNotNull(this)
            Assertions.assertEquals(roleName, role)
        }
    }
}
